### PR TITLE
docs: add a mention of uv for venvs near conda

### LINKED ISSUE #74: Add section on `uv` for advanced package resolution
Update `noxfile.py`, and maybe other areas with `uv`. 

- [x] The following lines at the top of a noxfile will cause nox to be many times faster if `uv` is installed. For example, a sphinx job takes 4 seconds instead of 22 seconds (that includes the time sphinx takes, so basically installation is nearly 0).
  ```python
  nox.needs_version = ">=2024.3.2"
  nox.options.default_venv_backend = "uv|virtualenv"
  ```
  see #71 

- [ ] We should also at least point people at `uv pip install`, as it's able to install anything in around 0 seconds (10-100x faster than pip). We could even cover how to use `uv pip compile` to lock your environment, though uv's `--exclude-newer` basically makes this no longer important, as you can recover a working build based on a date. It might be a bit advanced, but the `--resolution=lowest-direct` resolution strategy is also great for testing your minimum version limits. And `uv venv` can make a virtual environment in less time that it takes Python to start up.
  
   see #89 

_Originally posted by @henryiii in https://github.com/INTERSECT-training/packaging/issues/12#issuecomment-2142545146_

### PR DESCRIPTION
**Pull request by henryiii**
Mention `virtualenv` and `uv venv`.

- resolves #74

### CODE REVIEW

**Comment by henryiii** [comment]
Is there a way to make these boxes collapsable / collapsed by default? They really are "extra" information (especially the conda one) that would probably be better to opt-into by clicking them. IIRC there is often a way to do this, but it was missing from SWC?

**Review by hollandjg — COMMENTED** [review]
What do you think of this?

File: _episodes/01-environment.md [hunk 1]
```diff
@@ -76,6 +76,20 @@ The prompt will revert:
...
+> speed or update benefits. The installable `virtualenv` package provides the
+> same interface as the built-in `venv`, but is faster, has more options, and
+> has up-to-date embedded pip.
+>
+> Another alternative is `uv`, which is written in Rust, and `uv venv` is
+> actually faster than Python can even start up, though it doesn't install pip
+> by default (since you are supposed to use `uv pip`).
+>
+> Use the tool you prefer; the resulting venv works identically.
+{:.callout}
```
[the remark this answers is not in the document]
Comment (henryiii) [reply _episodes/01-environment.md:91]: But it's not a solution?

[the remark this answers is not in the document]
Comment (hollandjg) [reply _episodes/01-environment.md:91]: > But it's not a solution?

I think that's the only one of the special blockquotes which can collapse
https://carpentries.github.io/lesson-example/04-formatting/#special-blockquotes

[the remark this answers is not in the document]
Comment (hollandjg) [reply _episodes/01-environment.md:91]: > Unrelated, do we have support for tabs?

Like these tabs? https://facelessuser.github.io/pymdown-extensions/extensions/tabbed/#syntax-tab-2_1

I don't think so – I didn't see that anywhere in the documentation.

**Review by hollandjg — APPROVED** [review]
I think this is great!

File: _episodes/01-environment.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (henryiii) [reply _episodes/01-environment.md:91]: Yes, I'd like them for nox vs. hatch, for example.

[the remark this answers is not in the document]
Comment (henryiii) [reply _episodes/01-environment.md:91]: The thing it's based on did add collapsing to the rest of the blocks eventually from what I recall, SWC just seems frozen in time.

[the remark this answers is not in the document]
Comment (hollandjg) [reply _episodes/01-environment.md:91]: There's a new framework which some of the sessions have adopted this year: https://carpentries.github.io/sandpaper/

It's nicer, has inline instructor blocks you can hide, and is based around R-Markdown, but we didn't get round to implementing it.

**Comment by hollandjg** [comment]
Let's merge this as-is and we can look into hiding some content another time.

**Comment by henryiii** [comment]
Yes, that's a good future issue.
